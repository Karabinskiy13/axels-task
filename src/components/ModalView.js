import React from 'react';

import Modal from 'react-bootstrap/Modal';
import Image from 'react-bootstrap/Image';

const ModalView = ({ show, url, hideModal }) => {
  return (
    <Modal
      show={show}
      size="lg"
      fullscreen
      onHide={() => hideModal()}
      aria-labelledby="contained-modal-title-vcenter"
      centered>
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-vcenter">Image Preview</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Image src={url} fluid></Image>
      </Modal.Body>
    </Modal>
  );
};

export default ModalView;
