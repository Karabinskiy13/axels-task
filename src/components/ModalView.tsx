import React from 'react';

import { Modal, Image } from 'react-bootstrap';

type Props = {
  show: boolean;
  url: string;
  hideModal: () => void;
};

const ModalView = ({ show, url, hideModal }: Props) => (
  <Modal
    show={show}
    size="lg"
    fullscreen
    onHide={() => hideModal()}
    aria-labelledby="contained-modal-title-vcenter"
    centred>
    <Modal.Header closeButton>
      <Modal.Title id="contained-modal-title-vcenter">Image Preview</Modal.Title>
      <br />
    </Modal.Header>
    <Modal.Body style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
      <Image className="modal__image" src={url} fluid></Image>
    </Modal.Body>
  </Modal>
);

export default ModalView;
