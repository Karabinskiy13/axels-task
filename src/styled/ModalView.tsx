import styled from 'styled-components';
import { Box } from '@mui/material';

import { colors } from './Global';

export const StyledModal = styled(Box)`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  max-width: 1200px;
  max-height: 1000px;
  bgcolor: ${colors.bgColor};
  border: 2px solid ${colors.bgColor};
  box-shadow: 24;
  p: 4;
`;
