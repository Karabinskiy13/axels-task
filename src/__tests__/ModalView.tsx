import React from 'react';
import { render, screen } from '@testing-library/react';
import { jest } from '@jest/globals';

import ModalView from '../components/ModalView';

describe('<Modal View>', () => {
  test('Should render component', () => {
    const { asFragment } = render(
      <ModalView show={true} url={'/image.png'} hideModal={jest.fn()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Should display modal view from props', async () => {
    render(<ModalView show={true} url={'/image.png'} hideModal={jest.fn()} />);
    await screen.findAllByAltText('modal__image');
    expect(screen.getByAltText('modal__image')).toHaveAttribute('src', '/image.png');
  });
});
