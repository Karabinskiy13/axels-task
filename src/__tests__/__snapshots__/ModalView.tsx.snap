// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SinglePicture> Should render component 1`] = `
<Modal
  animation={true}
  aria-labelledby="contained-modal-title-vcenter"
  autoFocus={true}
  backdrop={true}
  centred={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  fullscreen={true}
  keyboard={true}
  onHide={[Function]}
  restoreFocus={true}
  show={true}
  size="lg"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      id="contained-modal-title-vcenter"
    >
      Image Preview
    </ModalTitle>
    <br />
  </ModalHeader>
  <ModalBody
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Image
      className="modal__image"
      fluid={true}
      rounded={false}
      roundedCircle={false}
      src="/image.png"
      thumbnail={false}
    />
  </ModalBody>
</Modal>
`;
